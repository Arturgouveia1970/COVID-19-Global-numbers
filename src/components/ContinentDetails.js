/* eslint-disable max-len */
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import CountUp from 'countup';
import images from '../images/images';
import CountryList from './CountryList';

const ContinentDetails = () => {
  const { continentName } = useParams();
  const continentDetails = useSelector((state) => state.covid.continents.find((cont) => cont.continent === continentName));
  const countries = useSelector((state) => state.covid.countries.filter(
    (country) => country.continent === continentName,
  ));

  return (
    <>
      {continentDetails && (
        <>
          <div className="continentDetails">
            <img
              src={images[continentName]}
              alt="continent map"
              className="continentImage"
            />
            <div className="continentCount">
              <span className="continentName">
                {continentDetails.continent}
              </span>
              <CountUp
                delay={1}
                end={continentDetails.cases}
                separator=","
                prefix="Total Cases: "
                duration={3}
              />
              <CountUp
                delay={1}
                end={continentDetails.deaths}
                separator=","
                prefix="Total Deaths: "
                duration={3}
              />
            </div>
          </div>
          <CountryList countries={countries} />
        </>
      )}
    </>
  );
};

export default ContinentDetails;
