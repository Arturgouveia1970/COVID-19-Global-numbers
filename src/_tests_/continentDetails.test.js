import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import ContinentDetails from '../components/ContinentDetails';

describe('snapshots testing', () => {
  test('snapshots for ContinentDetails component', () => {
    const rendererComponents = renderer
      .create(
        <Provider store={store}>
          <ContinentDetails />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponents).toMatchSnapshot();
  });
});
