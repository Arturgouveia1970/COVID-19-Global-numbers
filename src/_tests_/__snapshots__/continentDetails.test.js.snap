// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots testing snapshots for ContinentDetails component 1`] = `null`;
